### DISCUSSION TITLE
How to split big route file?

### DISCUSSION THREAD

**Discussion by ziimakc** [remix-run/remix#3063] [Q&A]
Let's say I have following structure:
```
routes
└── user(folder)
        ├── $userId.tsx(file)
                 ├── edit.tsx(file)
                 ├── index.tsx(file)
```

`index.tsx` is very big and I want to move some parts to other files, for example user delete component with form and action. 

I can possible move it out of routes folder and then import in `index.tsx` with action, but is there any way to do it with leaving it close to parent component?

```
routes
└── user(folder)
        ├── $userId.tsx(file)
                 ├── edit.tsx(file)
                 ├── index.tsx(file)
                 ├── del_component.tsx(file to import into index, not part of the routes)
```

**Comment by kiliman** [comment]
Check out the `remix-flat-routes` convention. It allows you to collocate supporting files with your routes. 

https://github.com/kiliman/remix-flat-routes
